File: docs/capture-screenshots.ts
import { chromium } from "playwright";
import path from "path";

const BASE_URL = "http://localhost:5733";
const OUT = path.join(__dirname, "screenshots");

async function main() {
  const browser = await chromium.launch({ headless: true });
  const context = await browser.newContext({
    viewport: { width: 1440, height: 900 },
    colorScheme: "dark",
  });
  const page = await context.newPage();

  // Give the app time to hydrate and load data
  console.log("1/8  Loading app...");
  await page.goto(BASE_URL, { waitUntil: "networkidle" });
  await page.waitForTimeout(3000);

  // --- Screenshot 1: Full sidebar with projects ---
  console.log("2/8  Sidebar overview...");
  await page.screenshot({ path: path.join(OUT, "01-sidebar-overview.png"), fullPage: false });

  // --- Screenshot 2: Jira import dropdown ---
  console.log("3/8  Jira import dropdown...");
  // Find the Jira ticket icon button in the sidebar header area
  const jiraButton = page.locator('button[aria-label*="Jira"], button:has(svg.lucide-ticket)').first();
  if (await jiraButton.isVisible()) {
    await jiraButton.click();
    await page.waitForTimeout(1500);
    await page.screenshot({ path: path.join(OUT, "02-jira-import.png"), fullPage: false });
    // Close the popover by pressing Escape
    await page.keyboard.press("Escape");
    await page.waitForTimeout(500);
  } else {
    console.log("   (Jira button not found, skipping)");
  }

  // --- Screenshot 3: Project context menu (right-click) ---
  console.log("4/8  Project context menu...");
  const projectItem = page.locator('[data-sidebar="menu-button"]').first();
  if (await projectItem.isVisible()) {
    await projectItem.click({ button: "right" });
    await page.waitForTimeout(800);
    await page.screenshot({ path: path.join(OUT, "03-context-menu.png"), fullPage: false });
    await page.keyboard.press("Escape");
    await page.waitForTimeout(500);
  } else {
    console.log("   (No project items found, skipping)");
  }

  // --- Screenshot 4: Command tray buttons ---
  console.log("5/8  Command tray...");
  // Click on the first project to open chat view with command tray
  const firstProject = page.locator('[data-sidebar="menu-button"]').first();
  if (await firstProject.isVisible()) {
    await firstProject.click();
    await page.waitForTimeout(1500);
    await page.screenshot({ path: path.join(OUT, "04-command-tray.png"), fullPage: false });
  }

  // --- Screenshot 5: SECDESK and Jira links on a project ---
  console.log("6/8  SECDESK/Jira links...");
  // Expand a project to show the SECDESK/Jira buttons
  const expandButton = page.locator('button:has-text("SECDESK"), button:has-text("Jira")').first();
  if (await expandButton.isVisible()) {
    // Scroll to it and screenshot the area
    await expandButton.scrollIntoViewIfNeeded();
    await page.waitForTimeout(500);
    // Take a cropped screenshot of the sidebar area
    const sidebar = page.locator('[data-sidebar="sidebar"]').first();
    if (await sidebar.isVisible()) {
      await sidebar.screenshot({ path: path.join(OUT, "05-secdesk-jira-links.png") });
    }
  } else {
    console.log("   (No SECDESK/Jira links visible, taking sidebar)");
    const sidebar = page.locator('[data-sidebar="sidebar"]').first();
    if (await sidebar.isVisible()) {
      await sidebar.screenshot({ path: path.join(OUT, "05-secdesk-jira-links.png") });
    }
  }

  // --- Screenshot 6: Standup modal ---
  console.log("7/8  Standup modal...");
  const standupButton = page.locator('button[aria-label*="tandup"], button:has(svg.lucide-message-square-text)').first();
  if (await standupButton.isVisible()) {
    await standupButton.click();
    await page.waitForTimeout(1000);
    await page.screenshot({ path: path.join(OUT, "06-standup-modal.png"), fullPage: false });
    await page.keyboard.press("Escape");
    await page.waitForTimeout(500);
  } else {
    console.log("   (Standup button not found, skipping)");
  }

  // --- Screenshot 7: Chat view with agent context ---
  console.log("8/8  Chat view...");
  await page.screenshot({ path: path.join(OUT, "07-chat-view.png"), fullPage: false });

  await browser.close();
  console.log(`\nDone! Screenshots saved to ${OUT}/`);
}

main().catch((err) => {
  console.error("Screenshot capture failed:", err);
  process.exit(1);
});

File: docs/capture-screenshots.py
#!/usr/bin/env python3
"""Capture screenshots of Fly Code features using Playwright."""

import os
from playwright.sync_api import sync_playwright

BASE_URL = "http://localhost:5733"
OUT = os.path.join(os.path.dirname(__file__), "screenshots")
os.makedirs(OUT, exist_ok=True)


def main():
    with sync_playwright() as p:
        browser = p.chromium.launch(headless=True)
        context = browser.new_context(
            viewport={"width": 1440, "height": 900},
            color_scheme="dark",
        )
        page = context.new_page()

        # Load app
        print("1/8  Loading app...")
        page.goto(BASE_URL, wait_until="networkidle")
        page.wait_for_timeout(3000)

        # 1: Full sidebar overview
        print("2/8  Sidebar overview...")
        page.screenshot(path=os.path.join(OUT, "01-sidebar-overview.png"))

        # 2: Jira import dropdown
        print("3/8  Jira import dropdown...")
        jira_btn = page.locator("button:has(svg.lucide-ticket)").first
        if jira_btn.is_visible():
            jira_btn.click()
            page.wait_for_timeout(2000)
            page.screenshot(path=os.path.join(OUT, "02-jira-import.png"))
            page.keyboard.press("Escape")
            page.wait_for_timeout(500)
        else:
            print("   (Jira button not found, trying alt selector)")
            # Try clicking by aria or text
            alt = page.locator('[aria-label="Import from Jira"]').first
            if alt.is_visible():
                alt.click()
                page.wait_for_timeout(2000)
                page.screenshot(path=os.path.join(OUT, "02-jira-import.png"))
                page.keyboard.press("Escape")
                page.wait_for_timeout(500)

        # 3: Project context menu
        print("4/8  Project context menu...")
        project = page.locator('[data-sidebar="menu-button"]').first
        if project.is_visible():
            project.click(button="right")
            page.wait_for_timeout(800)
            page.screenshot(path=os.path.join(OUT, "03-context-menu.png"))
            page.keyboard.press("Escape")
            page.wait_for_timeout(500)

        # 4: Click into a project to see chat + command tray
        print("5/8  Command tray...")
        # Click the first thread/project to open chat
        thread_link = page.locator('[data-sidebar="menu-sub-button"]').first
        if thread_link.is_visible():
            thread_link.click()
            page.wait_for_timeout(2000)
            page.screenshot(path=os.path.join(OUT, "04-chat-command-tray.png"))

        # 5: SECDESK and Jira links in sidebar
        print("6/8  SECDESK/Jira links...")
        sidebar = page.locator('[data-sidebar="sidebar"]').first
        if sidebar.is_visible():
            sidebar.screenshot(path=os.path.join(OUT, "05-sidebar-links.png"))

        # 6: Standup modal
        print("7/8  Standup modal...")
        standup_btn = page.locator("button:has(svg.lucide-message-square-text)").first
        if standup_btn.is_visible():
            standup_btn.click()
            page.wait_for_timeout(1000)
            page.screenshot(path=os.path.join(OUT, "06-standup-modal.png"))
            page.keyboard.press("Escape")
            page.wait_for_timeout(500)
        else:
            # Try by text
            alt = page.locator('button:has-text("Standup")').first
            if alt.is_visible():
                alt.click()
                page.wait_for_timeout(1000)
                page.screenshot(path=os.path.join(OUT, "06-standup-modal.png"))
                page.keyboard.press("Escape")
                page.wait_for_timeout(500)
            else:
                print("   (Standup button not found)")

        # 7: summary.md view
        print("8/8  Summary view...")
        summary_btn = page.locator('button:has-text("summary.md")').first
        if summary_btn.is_visible():
            summary_btn.click()
            page.wait_for_timeout(1000)
            page.screenshot(path=os.path.join(OUT, "07-summary-view.png"))
            page.keyboard.press("Escape")
            page.wait_for_timeout(500)

        # Final full-page shot
        page.screenshot(path=os.path.join(OUT, "08-full-app.png"))

        browser.close()
        print(f"\nDone! Screenshots saved to {OUT}/")


if __name__ == "__main__":
    main()
